core/src/main/java/com/datastax/oss/driver/internal/core/channel/ProtocolInitHandler.java

```diff
@@ -292,8 +292,14 @@ void onResponse(Message response) {
         } else {
           failOnUnexpected(response);
         }
-      } catch (AuthenticationException e) {
-        fail(e);
+      } catch (Exception e) {
+        if (e instanceof AuthenticationException) {
+          fail(e);
+        } else {
+          fail(
+              new AuthenticationException(
+                  channel.remoteAddress(), "Unexpected exception in encoutered in response ", e));
+        }
```

Comment:
I think two separate catch blocks would more idiomatic:
```java
} catch (AuthenticationException e) {
  fail(e);
} catch (Throwable t) {
  fail(
      new AuthenticationException(
          channel.remoteAddress(), "Unexpected exception in encoutered in response ", t));
}
```
(also, using Throwable for the generic one)